api4/cloud.go

```diff
@@ -121,6 +124,43 @@ func changeSubscription(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write(json)
 }
 
+func requestCloudTrial(c *Context, w http.ResponseWriter, r *http.Request) {
+	if c.App.Channels().License() == nil || !*c.App.Channels().License().Features.Cloud {
+		c.Err = model.NewAppError("Api4.requestCloudTrial", "api.cloud.license_error", nil, "", http.StatusInternalServerError)
```

Comment:
In a recent PR it was suggested that for non cloud instances that get cloud endpoint specific traffic, we should return `http.StatusForbidden`: https://github.com/mattermost/mattermost-server/pull/20152#discussion_r872013975

Reply:
ok